main.py

```diff
@@ -88,6 +88,16 @@ def knapsack_bqm(costs, weights, weight_capacity):
 
 sampler = LeapHybridSampler()
 sampleset = sampler.sample(bqm)
-solution = [df['weight'][i] for i in range(len(df['weight'])) if sampleset.record.sample[0][i] == 1.]
-
-print("Found solution {} at energy {}.".format(solution, sampleset.first.energy))
+for sample, energy in zip(sampleset.record.sample, sampleset.record.energy):
+
+    # Build solution from returned bitstring
+    solution = []
+    for this_bit_index, this_bit in enumerate(sample):
+        # The x's indicate whether each object has been selected
+        this_var = sampleset.variables[this_bit_index]
+        if this_bit == 1. and this_var.startswith('x'):
```

Comment:
Might be nicer as `if this_bit and .. :`, since we only expect `this_bit` to be 1 or zero.